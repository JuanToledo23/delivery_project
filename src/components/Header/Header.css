.header {
  padding: 25px 83px 25px 83px;
  background-color: white;
  z-index: 100;
}

.title-container {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  width: 100%;
}

.back {
  margin-right: 30px;
  cursor: pointer;
}

.homelogo {
  cursor: pointer !important;
  z-index: 1000;
}

.login-header {
  margin-right: 30px;
  z-index: 999;
}