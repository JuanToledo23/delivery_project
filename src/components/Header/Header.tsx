import './Header.css';
import Logo from 'assets/img/PALOIT.svg'
import { Link, useNavigate } from "react-router-dom";
import { ROUTES } from 'navigation/routes';
import { Login } from '@microsoft/mgt-react';

function Header() {
  
  const navigate = useNavigate();

  const logOut = () => {
    localStorage.removeItem("login");
    localStorage.removeItem("fullAccess");
    navigate('/');
  }

  return (
    <header className='header'>
      <div className='title-container'>
        <div className='decorative-bar'></div>
        <div className='login-header'>
          <Login logoutInitiated={() => logOut()} />
        </div>
        <Link to={ROUTES.home} className='homelogo'>
          <img src={Logo} alt='PALO IT Logo' data-testid="go_home" />
        </Link>
      </div>
    </header>
  );
}

export default Header;
