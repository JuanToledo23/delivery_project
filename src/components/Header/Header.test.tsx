import { render } from '@testing-library/react';
import Header from './Header';
import { BrowserRouter } from 'react-router-dom';

describe('Header', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});