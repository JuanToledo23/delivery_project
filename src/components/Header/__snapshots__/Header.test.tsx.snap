// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <div
      class="title-container"
    >
      <div
        class="decorative-bar"
      />
      <a
        class="homelogo"
        href="/home"
      >
        <img
          alt="PALO IT Logo"
          data-testid="go_home"
          src="PALOIT.svg"
        />
      </a>
    </div>
  </header>
</DocumentFragment>
`;
